.github/workflows/e2e.yml

```diff
@@ -17,16 +17,8 @@ jobs:
     timeout-minutes: 10
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.7.15
-      - name: Install
-        run: |
-          python -m pip install -U pip==22.3.1
-          python -m pip install -U setuptools==65.6.3
-          python -m pip install poetry==1.3.2
-          poetry config virtualenvs.create false
+      - name: Bootstrap
+        uses: ./.github/actions/bootstrap
```

Comment:
This implicitly changes Python from 3.7 to 3.8

Reply:
I'll update it to .17. 

As a sidenote in the improve_ci_speed PR there will be some more changes and explanations on that.

Reply:
It's often good to separate refactoring changes ("moving stuff around") and actual changes (changing how something behaves) into separate PR. A refactoring should not change any behaviour, which also makes it easier to verify that the refactoring is correct. The actual change (in this case: changing the version of those CI jobs) is trivial later on and should also be done in isolation. If we find an issue later on, this makes it easier to identify whether the issue has been introduced by the refactoring or by the version change.

TL;DR: let's stick with 3.7 in this PR

Reply:
Not sure I agree in this case as we are modifying the CI jobs but I'll revert the changes so each uses its own previous Python version and open PR's afterwards for each job.

Reply:
If we don't change the Python version, the DRY part of this PR is a typical refactoring. It just "moves code around", but the CI behaviour should not change. In fact, if it changes, it's a bug. The only small exception is the Pandoc installation because there the installation order changes.

Reply:
@danieljanes, all functional changes are reverted now.